compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -700,27 +700,41 @@ setVectorSizesForDynamicShapes(linalg::LinalgOp op,
                                VectorPreProcStrategy vecPreProcStrategy,
                                SmallVectorImpl<int64_t> &parallelSizes,
                                SmallVectorImpl<int64_t> &reductionSizes) {
-  // Masking doesn't need any dim set to 1.
-  if (vecPreProcStrategy == VectorPreProcStrategy::Masking) {
+  // TODO: Add support for ops with a mix of dynamic and static dims
+  if (!isFullyDynamicOp(op))
+    return;
+
+  if (vecPreProcStrategy != VectorPreProcStrategy::Peeling &&
```

Comment:
This is going to change the behavior for all the callers, including other targets, right?

To provide a bit more context, this function was introduced to control how to apply the `setAlwaysVectorizeSizes`, which basically sets tile sizes for dynamic dimensions to 1 so that vectorization doesn't bail out. In that sense, masking should never need any tile size to be set to one because it can handle dynamic dimensions using the mask. Same applies to other scenarios.

If your goal is to set those dimensions to one for SVE, I think you can do that where we first set the tile sizes for matmuls (e.g., `get*MatmulVectorSizes`). I guess I'm missing something because you changed that code already so, please, help me understand :)

Reply:
Thanks for this summary - I completely misread what this method is intended to do. Yup, I should be using `get*MatmulVectorSizes` instead :)